### PR TITLE
Improve dockerfile

### PR DESCRIPTION
**Pull request by QustodioPablo**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -13,7 +13,8 @@ services:
 
   api-gateway:
     build:
-      context: api_gateway
+      context: .
```
Comment (genismoreno) [inline docker-compose.yaml:16]: why the context switching?

File: api_gateway/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,11 @@
 FROM python:3.8
 
-COPY requirements.txt /
+COPY api_gateway/requirements.txt /
 RUN pip install -r requirements.txt
 
+COPY api_gateway /opt/api_gateway
```
Comment (genismoreno) [inline api_gateway/Dockerfile:6]: why do we need this in build time?

File: docker-compose.yaml (hunk 1 above)
Comment (QustodioPablo) [reply docker-compose.yaml:16]: Just for consistence, as the catalog is also using the root directory as its context.

File: api_gateway/Dockerfile (hunk 2 above)
Comment (QustodioPablo) [reply api_gateway/Dockerfile:6]: Without it, we would just have an empty python image. Even if the code is mounted as a volumen to speed up development (not having to rebuild to try changes), the code should still be in the image so we can use it once we want to deploy with kubernetes or just docker.

**Review by genismoreno — APPROVED** [review]

**Pull request merged**